runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/DataflowPipelineTranslator.java

```diff
@@ -936,8 +937,9 @@ private <K, V> void groupByKeyHelper(
                     && windowingStrategy.getWindowFn().assignsToOneWindow();
             if (isStreaming) {
               allowCombinerLifting &= transform.fewKeys();
-              // TODO: Allow combiner lifting on the non-default trigger, as appropriate.
-              allowCombinerLifting &= (windowingStrategy.getTrigger() instanceof DefaultTrigger);
+              // We don't currently have a good way to safely lift combiners in the face of count triggers, as we will
+              // lose track of how many elements have been combined.
+              allowCombinerLifting &= !containsCountTrigger(windowingStrategy.getTrigger());
```

Comment:
Any concerns about backwards compatibility here?

Reply:
I thing we should probably add a flag that allows disabling this behavior, and note in release notes that Dataflow update might require setting this flag for compatibility.

Reply:
On this note, we should probably have a policy about this. Historically I think we've generally required a flag for new behavior, such that we're backwards compatible by default. (But perhaps this doesn't impact as many users.) 

I'm on the fence here. Is it worth a query to dev@?

Reply:
This affects Dataflow only I think, so I don't think dev@ is the correct avenue here. Ideally Dataflow would make this decision in its backend, but not sure how easy that would be to do here.

Reply:
Yeah, ideally this could be done service-side...

Let's at least add a flag (TBD which way it should default).